cmd/deploy.go

```diff
@@ -404,6 +404,16 @@ func (dc *deployCmd) run() error {
 		return errors.Wrapf(err, "in SetPropertiesDefaults template %s", dc.apimodelPath)
 	}
 
+	//Validate image
+	cx, cancel := context.WithTimeout(context.Background(), armhelpers.DefaultARMOperationTimeout)
```

Comment:
1+ to ctx

Reply:
So in the deploy code, there is actually already a variable that is `ctx` which is a different context object. It is in light of this that I believe that the original author called it `cx` all I did was move the declaration of `cx` higher up. See DeployTemplate method which also uses the `cx`

Reply:
Ahh... interesting. I missed this.

You may have some unexpected side effects from this change. By moving the context higher up in the function and using it for **both** your calls for fetching images as well as the deployment of the template, you may introduce a a bug from the context timing out.

I would provide a new context for `ValidateDependencies()` which is the max amount of time to wait for that function to complete, and a second context for `DeployTemplate()`. This will separate the expiry of the two contexts.